### PR TITLE
Parse protocol/port in S3/Azure endpoints.

### PR DESCRIPTION
**Pull request by jreidthompson**
Utilize httpUrlNewParseP to parse endpoint and port from the url in the s3 and azure helpers to avoid issues where protocol was not expected to be part of the url.

https://github.com/pgbackrest/pgbackrest/issues/1560

### CODE REVIEW

**Review by dwsteele — CHANGES_REQUESTED** [review]
This looks pretty good, but I'd like to see some more tests updated with the new format.

File: src/storage/s3/helper.c [hunk 1]
```diff
@@ -24,14 +25,26 @@ storageS3Helper(const unsigned int repoIdx, const bool write, StoragePathExpress
...
-    const String *const endPoint = cfgOptionIdxHostPort(cfgOptRepoS3Endpoint, repoIdx, &port);
-    const String *const host = cfgOptionIdxHostPort(cfgOptRepoStorageHost, repoIdx, &port);
+    // Set the endpoint
+    const String *const endPoint = httpUrlHost(url);
 
-    // If the port option was set explicitly then use it in preference to appended ports
+    // Set the appended or default port
+    unsigned int port = httpUrlPort(url);
+
+    const String *host = NULL;
```
[the remark this answers is not in the document]
Comment (jreidthompson) [reply src/storage/s3/helper.c]: Corrected 79941de0345564d4e535031290832592e93eac25

File: src/storage/azure/helper.c [hunk 2]
```diff
@@ -21,27 +22,33 @@ storageAzureHelper(const unsigned int repoIdx, const bool write, StoragePathExpr
...
     {
-        endpoint = host;
+        endpoint = cfgOptionIdxStrNull(cfgOptRepoStorageHost, repoIdx);
 
         if (cfgOptionIdxSource(cfgOptRepoAzureUriStyle, repoIdx) == cfgSourceDefault)
             uriStyle = storageAzureUriStylePath;
     }
 
+    // if port was specified, overwrite the parsed/default port
+    if (cfgOptionIdxSource(cfgOptRepoStoragePort, repoIdx) != cfgSourceDefault)
```
[the remark this answers is not in the document]
Comment (jreidthompson) [reply src/storage/azure/helper.c:45]: Corrected c9a899338bd2fcfc2c8db319a117a446ace9bb01


Added various uri / uristyle tests also
177a12eca5af262d9c9a4db690c1fa78f094d642

**Review by dwsteele — APPROVED** [review]
Some formatting/comment updates in 644619a76f1bf39b21/5e35eb77cf2ba0b7c4.

**Pull request merged**